osbuild/main_cli.py

```diff
@@ -97,6 +98,9 @@ def osbuild_cli():
 
     index = osbuild.meta.Index(args.libdir)
 
+    # initialize error ids
+    OSbuildError.add_error_id(SourceException)
```

Comment:
This runs the risk of error IDs changing when we add more types.  One would have to know that error IDs are incremental to know that adding one somewhere else will change all the subsequent IDs.  Maybe it's obvious if you look at it for more than a few seconds, but not as obvious as a list of int literals in code.
It also makes it hard to add gaps between IDs in case we want to separate error codes into categories.
And then there's the matter of reverse lookups: if you want to find an error based on its ID, the ID never shows up in code and you have to basically count lines to figure out which is which.  Maybe not a realistic problem if they all have types and we only have a handful, but it's hard to know what we might need going forward.

Though an even bigger concern I have is whether this actually creates a persistent, single, global ID map (see comment in `formats_common.py`.

Reply:
I agree, I wanted to simplify the API to construct the list of error types, let's add a static number in front of every supported error. It'll be more clear.

Reply:
I went for an Enum way, I think this is better.